### ISSUE TITLE
Timesheet / List UI is very janky

### ISSUE THREAD

**Issue by Ali-Flt** [traggo/server#153]
**Describe the bug**
There are a couple of bugs:
1: You can't enter a two digit (higher than 9) hour manually.
2. when setting the end time manually, before completely entering the time, it sometimes changes the start time because it thinks that the end time is before the start time.

**To Reproduce**
1. Go to 'timesheet/list'
2. Add a manual time
3. Set the start time
4. Start setting the end time
5. See that the start time changes by itself

**Expected behavior**
Start time remains unchanged

**Comment by jmattheis** [comment]
> You can't enter a two digit (higher than 9) hour manually.

Yeah the date format is wrongly defined. You can configure the American24h, British or German date locale in the settings, they work fine.

> when setting the end time manually, before completely entering the time, it sometimes changes the start time because it thinks that the end time is before the start time.

This is kinda by design. The start date will be auto adjusted if the end date is valid and before the start date. But yeah, the user experience could be improved on that.

**Comment by zumoshi** [comment]
>  will be auto adjusted if the end date is valid and before the start date

I would prefer if it didn't. You can make it red or greyed out or something to let us know we can't save it until we change the other end of the pair.

But in my experience, it has always been a pain rather than helpful. Specially around AM/PM

E.g. I forgot to click end on time due to a distraction, so I wanted to edit the end time of the range to go from 12:10 pm to 11:50 am. I select AM and it suddenly thinks I finished at 10mins past midnight and pushes my start time which was correct to the previous day. and it auto saves it instantly without any undo. so now i have to remember what the other one was and change it back hoping it won't randomly also change my end time as well again.

It's bad enough that I've gotten to the habit of screenshotting the page any time I want to touch anything out of paranoia.
